src/test/java/StringManipulationTest.java

```diff
@@ -1,116 +1,197 @@
+/* Nadav Horowitz 6/3/2023 StringManipulationTest.java
```

Comment:
Test cases meaningfully cover all bounds and thrown exceptions, without any padded or trivial test cases.